File: MyAnimeListJava/src/main/java/com/example/myanimelistjava/models/enums/Genre.java
package com.example.myanimelistjava.models.enums;

public enum Genre {
    ADVENTURE("Adventure"),
    ACTION("Action"),
    COMEDY("Comedy"),
    SLICE_OF_LIFE("Slice of life"),
    DRAMA("Drama"),
    FANTASY("Fantasy"),
    SUPERNATURAL("Supernatural"),
    MAGIC("Magic"),
    MYSTERY("Mystery"),
    HORROR("Horror"),
    PSYCHOLOGICAL("Psychological"),
    SCI_FI("Sci-Fi"),
    ROMANCE("Romance");

    private final String value;

    Genre(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/models/enums/Genre.kt
package com.example.myanimelist.models.enums

import javafx.collections.FXCollections
import javafx.collections.ObservableList


enum class Genre(val value: String = "Unknown") {
    ADVENTURE("Adventure"),
    ACTION("Action"),
    COMEDY("Comedy"),
    SLICE_OF_LIFE("Slice of life"),
    DRAMA("Drama"),
    FANTASY("Fantasy"),
    SUPERNATURAL("Supernatural"),
    MAGIC("Magic"),
    MYSTERY("Mystery"),
    HORROR("Horror"),
    PSYCHOLOGICAL("Psychological"),
    SCI_FI("Sci-Fi"),
    ROMANCE("Romance");

    companion object {
        val sample: ObservableList<String> =
            FXCollections.observableArrayList(Genre.values().map { it.value }.toList())
    }
}